src/shared/generator.c

```diff
@@ -188,10 +188,50 @@ int generator_write_timeouts(
 
         return write_drop_in_format(dir, unit, 50, "device-timeout",
                                     "# Automatically generated by %s\n\n"
-                                    "[Unit]\nJobRunningTimeoutSec=%s",
+                                    "[Unit]\nJobTimeoutSec=%s",
                                     program_invocation_short_name, timeout);
 }
 
+int generator_write_device_deps(
+                const char *dir,
+                const char *what,
+                const char *where,
+                const char *opts) {
+
+        /* fstab records that specify _netdev option should apply the network
+         * ordering on the actual device depending on network connection. If we
+         * are not mounting real device (NFS, CIFS), we rely on _netdev effect
+         * on the mount unit itself. */
+
+        _cleanup_free_ char *node = NULL, *unit = NULL;
+        int r;
+
+        if (!fstab_test_option(opts, "_netdev\0"))
+                return 0;
+
+        node = fstab_node_to_udev_node(what);
+        if (!node)
+                return log_oom();
+
+        if (!is_device_path(node)) {
+                /* Nothing to apply dependencies to. */
+                return 0;
+        }
+
+        r = unit_name_from_path(node, ".device", &unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to make unit name from path: %m");
+
+        /* See mount_add_default_dependencies for explanation why we create such
+         * dependencies. */
+        return write_drop_in_format(dir, unit, 50, "netdev-dependencies",
+                                    "# Automatically generated by %1$s\n\n"
+                                    "[Unit]\nAfter=%2$s %3$s\nWants=%2$s",
+                                    program_invocation_short_name,
+                                    SPECIAL_NETWORK_ONLINE_TARGET,
+                                    SPECIAL_NETWORK_TARGET);
```

Comment:
nitpick: Note that SPECIAL_NETWORK_TARGET and SPECIAL_NETWORK_ONLINE_TARGET are macros resolving to literal strings. Hence I think we should write this as:

```c
        return write_drop_in_format(dir, unit, 50, "netdev-dependencies",
                                    "# Automatically generated by %1$s\n\n"
                                    "[Unit]\n"
                                    "After=" SPECIAL_NETWORK_ONLINE_TARGET " " SPECIAL_NETWORK_TARGET"\n"
                                    "Wants=" SPECIAL_NETWORK_ONLINE_TARGET "\n",
                                    program_invocation_short_name);
```

This would have the benefit that we'd format this string at compile time, instead of runtime.